src/coprocessor/dag/batch_executor/index_scan_executor.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use cop_datatype::EvalType;
+use kvproto::coprocessor::KeyRange;
+use tipb::expression::FieldType;
+use tipb::schema::ColumnInfo;
+
+use crate::storage::Store;
+
+use super::interface::*;
+use crate::coprocessor::codec::batch::{LazyBatchColumn, LazyBatchColumnVec};
+use crate::coprocessor::dag::expr::{EvalConfig, EvalContext};
+use crate::coprocessor::dag::Scanner;
+use crate::coprocessor::{Error, Result};
+
+pub struct BatchIndexScanExecutor<S: Store>(
+    super::scan_executor::ScanExecutor<
+        S,
+        IndexScanExecutorImpl,
+        super::ranges_iter::PointRangeConditional,
+    >,
+);
+
+impl<S: Store> BatchIndexScanExecutor<S> {
+    pub fn new(
+        store: S,
+        config: Arc<EvalConfig>,
+        columns_info: Vec<ColumnInfo>,
+        key_ranges: Vec<KeyRange>,
+        desc: bool,
+        unique: bool,
```

Comment:
Here `unique` means whether we can use point-get.
If unique is true, means it wants to scan a unique index with an exists key, exp:
`select * from t where col=1`
here `col` is a unique index, and the first step in `tidb` is using `indexscan` to find the `row_id`.

And if the unique is false, there are two situations:
1. scan from a non-unique index
2. scan from a unique index with `like`: `where unique-index like xxx`

Reply:
Thanks a lot for the supplement. I will add this to the comment.